garrysmod/lua/includes/modules/draw.lua

```diff
@@ -113,6 +113,62 @@ function SimpleTextOutlined(text, font, x, y, colour, xalign, yalign, outlinewid
 
 end
 
+--[[---------------------------------------------------------
+	Name: MultiColorText(font, x, y, xalign, yalign, vararg)
+	Desc: Simple draw text at position, but with multiple colors.
+	Usage: vararg is color tables with r/g/b/a elements followed by a string to be drawn with said color.
+-----------------------------------------------------------]]
+function MultiColorText( font, x, y, xAlign, yAlign, ... )
+	
+	font	= font		or "DermaDefault"
+	x		= x			or 0
+	y		= y			or 0
+	xAlign	= xAlign	or TEXT_ALIGN_LEFT
+	yAlign	= yAlign	or TEXT_ALIGN_TOP
+
+	surface.SetFont( font )
+
+	local ipairs, table, origin = ipairs({ ... })
+	local w, h = 0, 0
+
+	if ( xAlign != TEXT_ALIGN_LEFT ) then
+		local fullText = ""
+		
+		for k, v in ipairs, table, origin do
+			if not IsColor( v ) then
+				fullText = fullText .. tostring( v )

```

Comment:
This would be slow, using a table will be better, and even better if the table is localized outside the function.
But gmod is not that modern to use such things, so idk

Reply:
Using a table for what? This is used to find the length of the full text...
Do you mean for caching the lengths? Yeah it would be better but would that be worth it?

Reply:
table.concat that is

Reply:
separate concats are super slow

Reply:
table.concat doesn't `tostring()` the values, are you sure it'd be better to use it instead?

Reply:
Wait actually it'd be impossible to use it instead because we're expecting table values for the colors to be in that table. Or am I missing something here?

Reply:
He means create a separate table and add each tostringed entry, than table.concat it.